dexmaker/src/main/java/com/google/dexmaker/stock/ProxyBuilder.java

```diff
@@ -630,6 +631,43 @@ private static void check(boolean condition, String message) {
         for (MethodSetEntry entry : methodsToProxy) {
             results[i++] = entry.originalMethod;
         }
+
+        // Sort the results array so that they are returned by this method
+        // in a deterministic fashion.
+        Arrays.sort(results, new Comparator<Method>() {
+            @Override
+            public int compare(Method method1, Method method2) {
+                MethodId<? extends T, ?> methodId1 = getMethodId(method1);
```

Comment:
Another idea - is it possible to use a LinkedHashSet and call toArray() at the end?

It might not be possible since the "interfaces" could be specified in any order by the user.... although those could be sorted by name I guess.